app/email.go

```diff
@@ -593,6 +593,31 @@ func (es *EmailService) sendMailWithEmbeddedFiles(to, subject, htmlBody string,
 	return mailservice.SendMailWithEmbeddedFilesUsingConfig(to, subject, htmlBody, embeddedFiles, mailConfig, license != nil && *license.Features.Compliance, "")
 }
 
+func (es *EmailService) InvalidateVerifyEmailTokensForUser(userID string) *model.AppError {
+	tokens, err := es.srv.Store.Token().GetAllTokensByType(TokenTypeVerifyEmail)
```

Comment:
What about cheating here? Can we add a new method here, something like `GetAllTokensByTypeContainingString`, that gets the list of tokens by type, and filter them by `Extra LIKE '%string%'` and the string in this case is the userID. That would reduce the loop after, and would reduce the data fetching from the database.

I'm not sure if the email verify token is going to be a super common token in the database. Maybe "not optimizing" it have no impact at all.

Reply:
I think that not optimizing is fine here, but I think that as we're not controlling at all what goes in that extra data field, using simply a contains might cause unexpected side-effects later (just imagine that at some point we introduce a `RequesterId` as well)

Reply:
Maybe we can run a load-test to see the impact? (if it has any)

Reply:
If you add the `RequesterId` later, you end up fetching false positives, and filtering then like you are doing now (Actually, for me, we have to manually check that anyway). But the difference is you are not fetching the whole list, you reduce the amount of data requested and traveling.

Reply:
I'm ok leaving it as is now, but I think can be a problem eventually.